pkg/objstore/s3/s3.go

```diff
@@ -72,6 +72,9 @@ type Config struct {
 	// PartSize used for multipart upload. Only used if uploaded object size is known and larger than configured PartSize.
 	PartSize  uint64    `yaml:"part_size"`
 	SSEConfig SSEConfig `yaml:"sse_config"`
+
+	// Allow upstream callers to inject a round tripper
+	Transport http.RoundTripper `yaml:"-"`
```

Comment:
Hm, I would either inject it via some Setter or at least in HTTPConfig as there is functional overlap with that.

Reply:
That makes sense, I'll move it to the HTTPConfig